MapboxCoreNavigation/NavigationLocationManager.swift

```diff
@@ -21,8 +21,21 @@ open class NavigationLocationManager: CLLocationManager {
      Indicates whether the location manager’s desired accuracy should update
      when the battery state changes.
      */
+    @available(*, deprecated, message: "Manually override desiredAccuracy instead.")
     public var automaticallyUpdatesDesiredAccuracy = true
     
+    var accuracyOverride: CLLocationAccuracy?
+    
+    override open var desiredAccuracy: CLLocationAccuracy {
+        get {
+            if let override = accuracyOverride { return override }
+            return  UIDevice.current.isPluggedIn ? kCLLocationAccuracyBestForNavigation : kCLLocationAccuracyBest
+        }
+        set {
+            accuracyOverride = newValue
```

Comment:
What’s the purpose of this override? Now if client code sets `NavigationLocationManager.desiredAccuracy`, the actual `CLLocationManager.desiredAccuracy` is unaffected, but `NavigationLocationManager.desiredAccuracy` lies; `desiredAccuracy` shadows the actual value.